worklist/views.py

```diff
@@ -1,36 +1,78 @@
 import json
-from django.http import JsonResponse
+from django.http import JsonResponse, HttpResponseRedirect
 from django.shortcuts import render
 import logging
 from worklist.constants import SEARCH_BY_NAME_FOR_WORKLISTS, \
-    SEARCH_BY_USERNAME_FOR_WORKLISTS
+    SEARCH_BY_USERNAME_FOR_WORKLISTS, ARTICLE_STATUS_TO_NUMBER_MAPPING, \
+    ARTICLE_NUMBER_TO_STATUS_MAPPING
 from worklist.views_helper_functions.store_articles import \
     store_added_articles, store_psid_articles
 from worklist.models import WorkList, Task
+from django.contrib.auth.decorators import login_required
+from django.contrib.auth import logout
+
 
 logger = logging.getLogger('django')
 
 
+# View related to OAuth using MediaWiki
+def mediawiki_login(request):
+    return HttpResponseRedirect(
+        'https://tools.wmflabs.org/worklist-tool/oauth/login/mediawiki/')
+
+
+# View for logging out from the app
+@login_required
+def app_logout(request):
+    logout(request)
+    # if user has logged out of the app, take him to home page
+    message = ''
+    is_logged_in = False
+    error = False
```

Comment:
Nit: no need for the temporary variables here as the dictionary keys below already tell the reader what the values mean, so it's just as easy to read:

```python
return render(request, 'homepage.html', {'message': '', 'is_logged_in': False, ...})
```

Reply:
Done.